src/CompileTime.hx

```diff
@@ -64,6 +70,17 @@ class CompileTime
         return toExpr(obj);
     }
 
+    #if yaml
+    macro public static function parseYamlFile(path:String) {
+      var content = loadFileAsString(path);
+      var data = Yaml.parse(content, Parser.options().useObjects());
+      var s = haxe.Json.stringify(data);
+      var json = haxe.Json.parse(s);
+      return toExpr(json);
```

Comment:
I thought there must be a good reason. Thanks again!
On 30 Nov 2015 8:52 am, "francesco agati" <notifications@github.com> wrote:

> In src/CompileTime.hx
> <https://github.com/jasononeil/compiletime/pull/6#discussion_r46098577>:
>
> > @@ -64,6 +70,17 @@ class CompileTime
> >          return toExpr(obj);
> >      }
> >
> > +    #if yaml
> > +    macro public static function parseYamlFile(path:String) {
> > +      var content = loadFileAsString(path);
> > +      var data = Yaml.parse(content, Parser.options().useObjects());
> > +      var s = haxe.Json.stringify(data);
> > +      var json = haxe.Json.parse(s);
> > +      return toExpr(json);
>
> var data = Yaml.read(file, Parser.options().useObjects());
> //var s = haxe.Json.stringify(data);
> //var json = haxe.Json.parse(s);
>
> return Context.makeExpr(data, Context.currentPos());
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jasononeil/compiletime/pull/6/files#r46098577>.
>